### ISSUE TITLE
Is it possible to use customer service conversation as data set with this model ?

### ISSUE THREAD

**Issue by genz10** [DongjunLee/conversation-tensorflow#5]
Hi,

Is it possible to use customer service conversation as data set with this model ? such as

```
Hello
May I help you ?
I want to ask something
Yes please
Does my booking refundable ?
May I get your booking ID.
1234567890
Yes, your booking is refundable.
I want to submit a refund for my booking.
...
...
...
```

**Comment by DongjunLee** [comment]
Hi @genz10,
Yes it is.

If you make a chatbot that automates custom services, you need lots of customer service conversation data. And there will be uncertainty in the answer.

**Comment by genz10** [comment]
@DongjunLee 

Thanks for your reply. FYI I've been successfully reproduce tiny model, after 20K epoch and final loss 0.0223939. I found that, when the epoch is not enough it will caused loss value to be unstable.

For now, I'm still experimenting your model to be applied into customer service data set. (Arround 8K question and 8K answer), should I report the result here ?

**Comment by DongjunLee** [comment]
I think 8K data set is not enough, but maybe it can be working.
I wonder what your experiment results are. :)

**Comment by genz10** [comment]
How this model overcome such conversation like this

```
A : Hello
A : I want to ask something
A : Regarding my purchase yesterday
B : Yes please
```

There is multiple question before there is an answer.
 
My idea is to concatenate multiple sentences into 1 line. But will it works ?

**Comment by DongjunLee** [comment]
I think that simply solution makes the input sentences are long and it means Increase data sparsity. 
But you need to try it.

Or check this paper : [Building End-To-End Dialogue Systems Using Generative Hierarchical Neural Network Model](https://arxiv.org/pdf/1507.04808.pdf)

**Comment by genz10** [comment]
I've checked that paper, and gives me some idea to use triplets model instead of single sentence for each line.

But for the training, after reach 20k epoch and 0,02XXXX loss value, I found that the model can't answer simple question, I'm still investigating what's wrong.

**Comment by DongjunLee** [comment]
Maybe that is because of lack of data, it seems to be overfitting to the training data.
If you want to get the same results as training data, try doing the beam_width as 0. (using GreedyEmbeddingHelper)
